tools/bufr/QueryPrinter/NcepQueryPrinter.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+ * (C) Copyright 2022 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "NcepQueryPrinter.h"
+
+#include <memory>
+#include <iostream>
+#include <sstream>
+
+#include "eckit/exception/Exceptions.h"
+
+#include "QueryPrinter.h"
+#include "../../../src/bufr/BufrParser/Query/DataProvider/NcepDataProvider.h"
+#include "../../../src/bufr/BufrParser/Query/SubsetTable.h"
+
+namespace Ingester {
+namespace bufr {
+
+    NcepQueryPrinter::NcepQueryPrinter(const std::string& filepath) :
+      QueryPrinter(std::make_shared<NcepDataProvider>(filepath))
+    {
+    }
+
+    std::vector<QueryData> NcepQueryPrinter::getQueries(const SubsetVariant& variant)
+    {
+        if (dataProvider_->isFileOpen())
+        {
+            std::ostringstream errStr;
+            errStr << "Tried to call QueryPrinter::getQueries, but the file is already open!";
+            throw eckit::BadParameter(errStr.str());
```

Comment:
Why would it be a problem if the file is already open?

Reply:
You can not open a file twice. The operating system would not permit it. We would have an exception... this just provides a nicer error message.

Reply:
Understood.